battleship.py

```diff
@@ -183,15 +203,26 @@ def drawShip(data, canvas, ship):
 Returns: bool
 '''
 def shipIsValid(grid, ship):
-    return
-
+    if len(ship) == 3:
+        if checkShip(grid, ship):
+            if isVertical(ship):
+                return True
+            elif isHorizontal(ship):
+                return True
+    return False
 
 '''
 placeShip(data)
 Parameters: dict mapping strs to values
 Returns: None
 '''
 def placeShip(data):
+    if shipIsValid(data["userGrid"], data["tempShips"]) :
+        for j in range(len(data["tempShips"])):
+                addShips(data["userGrid"], data["numShips"])
+                data["numUserShips"]+=1
+    else:
+        print("Error: Ship is not Valid")
     return

```

Comment:
@Sunitha101199 Please check line number 223 in which increment should be inside the loop. After updating the value to SHIP_UNCLICKED, then you can increment.